src/storage/kv_storage/littlefs/kv_littlefs.cpp

```diff
@@ -0,0 +1,143 @@
+/**
+ * This software is distributed under the terms of the MIT License.
+ * Copyright (c) 2023 LXRobotics.
+ * Author: Alexander Entinger <alexander.entinger@lxrobotics.com>
+ * Contributors: https://github.com/107-systems/107-Arduino-Cyphal-Support/graphs/contributors.
+ */
+
+/**************************************************************************************
+ * INCLUDE
+ **************************************************************************************/
+
+#include "kv_littlefs.h"
+
+#if __GNUC__ >= 11
+
+#include <map>
+#include <string>
+#include <variant>
+#include <sstream>
+
+/**************************************************************************************
+ * NAMESPACE
+ **************************************************************************************/
+
+namespace cyphal::support::platform::storage
+{
+
+/**************************************************************************************
+ * CONSTANTS
+ **************************************************************************************/
+
+static std::map<littlefs::Error, Error> const LITTLEFS_TO_STORAGE_ERROR_MAP =
+{
+  {littlefs::Error::IO          , Error::IO},
+  {littlefs::Error::CORRUPT     , Error::Internal},
+  {littlefs::Error::NOENT       , Error::Existence},
+  {littlefs::Error::EXIST       , Error::Existence},
+  {littlefs::Error::NOTDIR      , Error::Internal},
+  {littlefs::Error::ISDIR       , Error::API},
```

Comment:
These are closer to the existence error, as we found a file where there should have been a directory and the other way around. Otherwise, case-by-case handling is probably required.